### PR TITLE
Fix build when adding a new Problem

### PR DESCRIPTION
**Pull request by A-Hilaly**
#161

### CODE REVIEW

**Comment by A-Hilaly** [comment]
Thanks (the code was mine x)) ! the build is passing normaly. I changed exit status when there are no added solutions like this PR.

File: stats.py [hunk 1]
```diff
@@ -507,11 +509,7 @@ def handle_files(files):
     for f in files:
         if f.count("/") == 2:
             solutions.append(tuple(f.split("/")))
-        elif f.count("/") == 1 and "Problem" in f:
-            f = f.strip('/')
-            solutions += [(f, lang, solution) for lang in os.listdir(os.getcwd() + '/' + f)
```
Comment (ryukinix) [inline stats.py]: Removing that section of code, how the function `handle_files` can decide if the solution file language is in the BUILD_SUPPORT?

I don't understand why are  you adding files listed in BUILD_FILES. That will be ignored after filtering by is_solution function. Why this?

Comment (A-Hilaly) [reply stats.py]: the function `build_results` decide if the solution file language is in BUILD_SUPPORT or not
 
When I first wrote this function I didn't known how git diff works so I wrote a generalist function. Basically when there is only one `/` it means that we have a probably a directory like "ProblemX/Y" , and what this function was doing is collecting all solutions inside Y ( thats exactly the bug, it was trying to find solutions inside ./Readme and .hash) .
I could have simply added a 
```python
if 'README' in f or '.hash' in f :
     continue
```
`git diff origin/master --name-only` already generates to us file name with there associated paths starting from $PWD
in this case paths with ProblemX/Y/solutionZ will be directly treated in the case of `f.count("/") == 2`.
(This feature could be useful if wanted to build all solutions inside a /Problem/LangX directory)

Comment (A-Hilaly) [reply stats.py]: > I don't understand why are you adding files listed in BUILD_FILES. That will be ignored after filtering by is_solution function. Why this?

BUILD_FILES are not added to solutions list. They are send to build_files which contain a list modified build files (like stats.py). I I remember when this case was generating a build of all solutions across all languages. with target this build files we can avoid a type of building when update the `README.md` or `.gitignore` for example

Comment (ryukinix) [reply stats.py]: Ok, I'll merge this for now. You test that solution, right?

Comment (A-Hilaly) [reply stats.py]: I tested only on my machine. Ill PR soon to test if this is working with travis !

**Pull request merged**